### PR TITLE
Fix crash hit during PR's fuzzing job

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alan.jowett@microsoft.com>

## Description

Validate that a map operation is supported on a specific map type.

## Testing

CI/CD & fuzzing.

## Documentation

No

### CODE REVIEW

File: libs/execution_context/unit/execution_context_unit_test.cpp [hunk 1]
```diff
@@ -222,6 +222,13 @@ _test_crud_operations(ebpf_map_type_t map_type)
     auto retrieved_map_definition = *ebpf_map_get_definition(map.get());
     retrieved_map_definition.value_size = ebpf_map_get_effective_value_size(map.get());
     REQUIRE(memcmp(&retrieved_map_definition, &map_definition, sizeof(map_definition)) == 0);
+
+    // Negative test for key size
```
Comment (dthaler) [inline libs/execution_context/unit/execution_context_unit_test.cpp:226]:
```suggestion
    // Negative test for key size.
```

File: libs/execution_context/unit/execution_context_unit_test.cpp [hunk 2]
```diff
@@ -700,4 +740,32 @@ TEST_CASE("ring_buffer_async_query", "[execution_context]")
...
\ No newline at end of file
+
+    {
+        uint32_t key = 0;
+        uint32_t value2 = 0;
+        REQUIRE(
+            ebpf_map_update_entry(map.get(), sizeof(key), reinterpret_cast<uint8_t*>(&key), 0, nullptr, EBPF_ANY, 0) ==
+            EBPF_INVALID_ARGUMENT);
+
+        // Negative test cases
```
Comment (dthaler) [inline libs/execution_context/unit/execution_context_unit_test.cpp:751]:
```suggestion
        // Negative test cases.
```

**Review by dthaler — APPROVED** [review]

File: libs/execution_context/ebpf_maps.c [hunk 3]
```diff
@@ -2125,6 +2129,16 @@ ebpf_map_delete_entry(_In_ ebpf_map_t* map, size_t key_size, _In_reads_(key_size
...
         return EBPF_INVALID_ARGUMENT;
     }
+
+    if (ebpf_map_metadata_tables[map->ebpf_map_definition.type].delete_entry == NULL) {
+        EBPF_LOG_MESSAGE_UINT64(
+            EBPF_TRACELOG_LEVEL_ERROR,
+            EBPF_TRACELOG_KEYWORD_MAP,
+            "ebpf_map_delete_entry not supported on map",
+            map->ebpf_map_definition.type);
+        return EBPF_INVALID_ARGUMENT;
```
Comment (saxena-anurag) [inline libs/execution_context/ebpf_maps.c:2139]: minor: should this be `EBPF_OPERATION_NOT_SUPPORTED` instead?

**Review by saxena-anurag — APPROVED** [review]

File: libs/execution_context/ebpf_maps.c (hunk 3 above)
Comment (Alan-Jowett) [reply libs/execution_context/ebpf_maps.c]: @saxena-anurag fixed it, can you please re-approve?

**Review by saxena-anurag — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**